NitroxClient/Debuggers/EntityDebugger.cs

```diff
@@ -18,24 +26,93 @@ private void RenderEntityDebugger()
 
         public override void OnGUI()
         {
+            if (!Enabled)
+            {
+                return;
+            }
+
+            rects.Clear();
             foreach (KeyValuePair<NitroxId, GameObject> gameObjectPairs in NitroxEntity.GetGameObjects())
             {
                 NitroxId id = gameObjectPairs.Key;
                 GameObject gameObject = gameObjectPairs.Value;
-                if (gameObject != null && gameObject != Player.mainObject)
+                if (gameObject == null || gameObject == Player.mainObject)
+                {
+                    continue;
+                }
+
+                Vector3 screenPos = Player.main.viewModelCamera.WorldToScreenPoint(gameObject.transform.position);
+                if (screenPos.z > 0 && screenPos.z < 20 &&
+                    screenPos.x >= 0 && screenPos.x < Screen.width &&
+                    screenPos.y >= 0 && screenPos.y < Screen.height)
                 {
-                    Vector3 screenPos = Player.main.viewModelCamera.WorldToScreenPoint(gameObject.transform.position);
-                    if (screenPos.z > 0 && screenPos.z < 20 &&
-                        screenPos.x >= 0 && screenPos.x < Screen.width &&
-                        screenPos.y >= 0 && screenPos.y < Screen.height)
+                    GUIStyle style = GUI.skin.label;
+                    GUIContent textContent = new GUIContent("ID " + id.ToString() + "   NAME " + gameObject.name);
+                    Vector2 size = style.CalcSize(textContent);
+                    size += new Vector2(10f, 0f); //for box edges
+
+                    Vector2 pointLocation = new Vector2(screenPos.x, Screen.height - screenPos.y);
+                    Rect drawSize = new Rect(screenPos.x + textXOffset, Screen.height - screenPos.y, size.x, size.y);
+                    while (true)
                     {
-                        GUIStyle style = GUI.skin.label;
-                        GUIContent textContent = new GUIContent("ID " + id.ToString());
-                        Vector2 size = style.CalcSize(textContent);
-                        GUI.Label(new Rect(screenPos.x, Screen.height - screenPos.y, size.x, size.y), textContent);
+                        bool finished = true;
+                        foreach (Rect rect in rects)
+                        {
+                            if (rect.Overlaps(drawSize))
+                            {
+                                drawSize.x = rect.x;
```

Comment:
with this line: https://cdn.discordapp.com/attachments/528915769107415040/689549996638339197/unknown.png
without this line: https://cdn.discordapp.com/attachments/528915769107415040/689550007598448643/unknown.png

not sure whether to keep or remove, tell me your thoughts

Reply:
Think the first will give a bit more "organized mess" of NitroxIDs :D

Reply:
I'm for the first too